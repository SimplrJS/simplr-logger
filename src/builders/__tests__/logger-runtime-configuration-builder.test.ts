import { LoggerRuntimeConfigurationBuilder, LoggerConfigurationBuilder, LogLevel } from "simplr-logger";

test("Default Configuration", () => {
    const runtimeConfiguration = new LoggerRuntimeConfigurationBuilder();
    runtimeConfiguration.UpdateConfiguration(builder => builder.SetPrefix("New prefix").Build());
    expect(runtimeConfiguration).toMatchSnapshot();
});

test("Same Configuration", () => {
    const config = new LoggerConfigurationBuilder().SetDefaultLogLevel(LogLevel.Trace).Build();
    const runtimeConfiguration = new LoggerRuntimeConfigurationBuilder(config);
    runtimeConfiguration.UpdateConfiguration(builder => builder.Build());
    expect(runtimeConfiguration).toMatchSnapshot();
});

test("Custom Configuration", () => {
    const config = new LoggerConfigurationBuilder().SetPrefix("Old prefix").Build();
    const runtimeConfiguration = new LoggerRuntimeConfigurationBuilder(config);
    runtimeConfiguration.UpdateConfiguration(builder => builder.SetPrefix("New prefix").Build(), true);
    expect(runtimeConfiguration).toMatchSnapshot();
});

test("New Configuration", () => {
    const runtimeConfiguration = new LoggerRuntimeConfigurationBuilder();
    runtimeConfiguration.UpdateConfiguration(builder => builder.Build(), false);
    expect(runtimeConfiguration).toMatchSnapshot();
});
