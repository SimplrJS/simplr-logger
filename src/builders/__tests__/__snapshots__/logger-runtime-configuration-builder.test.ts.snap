// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Configuration 1`] = `
LoggerRuntimeConfigurationBuilder {
  "Configuration": Object {
    "DefaultLogLevel": Object {
      "LogLevel": 4,
      "LogLevelIsBitMask": false,
    },
    "Prefix": "New prefix",
    "WriteMessageHandlers": Array [
      Object {
        "Handler": ConsoleMessageHandler {
          "configuration": Object {
            "LogLevelPrefix": "short",
            "TimePrefix": "short",
            "UseColors": false,
          },
          "defaultConfiguration": Object {
            "LogLevelPrefix": "short",
            "TimePrefix": "short",
            "UseColors": false,
          },
        },
        "LogLevel": 4,
        "LogLevelIsBitMask": false,
      },
    ],
  },
}
`;

exports[`Default Configuration 1`] = `
LoggerRuntimeConfigurationBuilder {
  "Configuration": Object {
    "DefaultLogLevel": Object {
      "LogLevel": 4,
      "LogLevelIsBitMask": false,
    },
    "Prefix": "New prefix",
    "WriteMessageHandlers": Array [
      Object {
        "Handler": ConsoleMessageHandler {
          "configuration": Object {
            "LogLevelPrefix": "short",
            "TimePrefix": "short",
            "UseColors": false,
          },
          "defaultConfiguration": Object {
            "LogLevelPrefix": "short",
            "TimePrefix": "short",
            "UseColors": false,
          },
        },
        "LogLevel": 4,
        "LogLevelIsBitMask": false,
      },
    ],
  },
}
`;

exports[`Same Configuration 1`] = `
LoggerRuntimeConfigurationBuilder {
  "Configuration": Object {
    "DefaultLogLevel": Object {
      "LogLevel": 32,
      "LogLevelIsBitMask": false,
    },
    "Prefix": undefined,
    "WriteMessageHandlers": Array [
      Object {
        "Handler": ConsoleMessageHandler {
          "configuration": Object {
            "LogLevelPrefix": "short",
            "TimePrefix": "short",
            "UseColors": false,
          },
          "defaultConfiguration": Object {
            "LogLevelPrefix": "short",
            "TimePrefix": "short",
            "UseColors": false,
          },
        },
        "LogLevel": 32,
        "LogLevelIsBitMask": false,
      },
    ],
  },
}
`;
